.github/workflows/integration-tests.yml

```diff
@@ -123,11 +123,18 @@ jobs:
         with:
           name: coverage-artifacts-html
           path: '/var/tmp/report'
-      
+
       - name: Upload coverage file
         if: always()
         uses: actions/upload-artifact@v3
         with:
           name: coverage-artifacts-info
           path: '/var/tmp/merged.info'
-      
\ No newline at end of file
+
+      - name: Report to Coveralls
+        if: always()
```

Comment:
Shouldn't we update coveralls info only when:

1. PR is merged
2. Whole run was successfull

?

Reply:
Regarding (1), no. Reporting to Coveralls in the PR allows it to add comments back (e.g. block PR if the cover rate drops below a certain level). When the PR is merged to `main`, the entire workflow is executed again (per its configuration) and then the cover rate for `main` is reported (and can be reflected in a README badge)
Regarding (2), I'm not sure

Reply:
To 1.:
We'd need to figure that out. If I understood it correctly, coveralls does distinguish between branches/forks - so all we have to do is focus on the main branch. Also, we'd probably want to get feedback if coverage drops when the PR is still open - to mitigate early. 

To 2.:
Yes, the whole collecting and uploading coverage is a bit brittle. For example, if no test was run (e.g. infra issues), we get quite obfuscated logs since the `create_coverage_report.py` in the finish step also fails. We can tackle it here or improve later on. 
In GH workflows its a bit tricky to check if a file exists on that level. I only know an approach to check if the hash is empty, like this:
```
if: ${{ hashFiles('merged.info') != "" }}
```